src/js/integrations/reactnativeerrorhandlers.ts

```diff
@@ -58,15 +58,28 @@ export class ReactNativeErrorHandlers implements Integration {
     if (this._options.onunhandledrejection) {
       /*
         In newer RN versions >=0.63, the global promise is not the same reference as the one imported from the promise library.
-        This is due to a version mismatch between promise versions. The version will need to be fixed with a package resolution.
-        We first run a check and show a warning if needed.
+        This is due to a version mismatch between promise versions.
       */
-      this._checkPromiseVersion();
+      this._polyfillPromise();
+    }
+  }
+  /**
+   * Polyfill the global promise instance with one we can be sure that we can attach the tracking to.
+   */
+  private _polyfillPromise(): void {
+    /* eslint-disable import/no-extraneous-dependencies,@typescript-eslint/no-var-requires*/
+    const {
+      polyfillGlobal,
+    } = require("react-native/Libraries/Utilities/PolyfillFunctions");
+    const Promise = require("promise/setimmediate/es6-extensions");
 
+    require("promise/setimmediate/done");
+    require("promise/setimmediate/finally");
+
+    polyfillGlobal("Promise", () => {
       const tracking: {
```

Comment:
Is it fine that the require of `promise/setimmediate/es6-extensions` happens outside the function closure?

Are we in danger of other people mutating/patching the promise export?